Replace the lengthy header for license in all source files with a shorter header

chihtaw opened the pull request:
Replace with the following header:

```
// SPDX-License-Identifier: GPLv3
// SPDX-FileCopyrightText: 2024 OGL authors
```

chihtaw committed:
Replace the lengthy header for license in all source files with
a shorter header.

chihtaw committed:
Remove old workflow files
Add a new workflow file

greole commented on .github/workflows/static_checks.yaml:
```diff
@@ -0,0 +1,129 @@
+
+    - name: Create Compilation Database
+      run: |
+        cmake --preset ninja-cpuonly-all \
+          -DNEOFOAM_DEVEL_TOOLS=OFF \
+          -DCMAKE_CXX_COMPILER=clang++-16 \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+          -DNEOFOAM_ENABLE_IWYU=ON
+        cmake --build --preset ninja-cpuonly-all
```
```suggestion
        cmake --preset ninja-cpuonly-release\
        -DCMAKE_CXX_COMPILER=clang++-16 \
        -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
        cmake --build --preset ninja-cpuonly-release
```

chihtaw committed:
Add copyright and license informatin to all files with the file
extension .C or .H

chihtaw committed:
Edit the log file to include the change:
Switched to the SPDX headers

chihtaw committed:
Add the copyright and licensing info to all the remaining files which
do not have their copyright and licensing marked yet.

Add some license files for some uncommentable files.

Add the .reuse directory generated by the reuse tool for licensing

greole commented on CMakePresets.json.license:
Please use the .reuse/dep5 file instead.

greole commented on Doxyfile.license:
Please use the .reuse/dep5 file instead.

greole commented on common/.common.H.swo.license:
Please remove this file: common/.common.H.swo

greole commented on test/channel_validation.json.license:
Please use the .reuse/dep5 file instead.

greole commented on test/validation.json.license:
Please use the .reuse/dep5 file instead.

greole reviewed (commented):
Some quick comments.

chihtaw committed:
Remove all .license files.
Add the copyright and licensing info to the ./reuse/dep5 file for
individual uncommentable files.

chihtaw committed:
Remove .license file for .github/workflows/test_matrix.json
Add the corresponding copyright and licensing info to the .reuse/dep5 file

chihtaw committed:
Use clang-format to format the following files:
- DevicePersistent/Array/Array.H
- DevicePersistent/ExecutorHandler/ExecutorHandler.H

chihtaw committed:
Update .github/workflows/static_checks.yaml

greole commented on _typos.toml:
```diff
@@ -1,3 +1,7 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
 [files]
 extend-exclude = ["third_party/*"]
```
```suggestion
extend-exclude = ["third_party/*", "Doxyfile"]
```

greole committed:
Update _typos.toml

greole commented on .github/workflows/static_checks.yaml:
```diff
@@ -0,0 +1,131 @@
+      run: |
+        cmake --preset ninja-cpuonly-release\
+        -DCMAKE_CXX_COMPILER=clang++-16 \
+        -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
```
```suggestion
        -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```

chihtaw committed:
Update log file for changes

greole commented on .github/workflows/static_checks.yaml:
```diff
@@ -0,0 +1,131 @@
+
+    - name: Create Compilation Database
+      run: |
+        cmake --preset ninja-cpuonly-release\
```
```suggestion
        cmake --preset ninja-cpuonly-release \
```

greole reviewed (approved):
LGTM!
